image_classification/inception_v4.py

```diff
@@ -0,0 +1,526 @@
+        stride=1,
+        padding=1)
+    conv2 = img_conv(
+        name='stem_conv2',
+        input=conv1,
+        num_channels=32,
+        num_filters=64,
+        filter_size=3,
+        stride=1,
+        padding=1)
+
+    def block0(input):
+        pool0 = paddle.layer.img_pool(
+            name='stem_branch0_pool0',
+            input=input,
+            num_channels=64,
+            pool_size=3,
+            stride=2,
+            pool_type=paddle.pooling.Max())
+        conv0 = img_conv(
+            name='stem_branch0_conv0',
+            input=input,
+            num_channels=64,
+            num_filters=96,
+            filter_size=3,
+            stride=2,
+            padding=1)
+        return paddle.layer.concat(input=[pool0, conv0])
+
+    def block1(input):
+        l_conv0 = img_conv(
+            name='stem_branch1_l_conv0',
+            input=input,
+            num_channels=160,
+            num_filters=64,
+            filter_size=1,
+            stride=1,
+            padding=0)
+        l_conv1 = img_conv(
+            name='stem_branch1_l_conv1',
+            input=l_conv0,
+            num_channels=64,
+            num_filters=96,
+            filter_size=3,
+            stride=1,
+            padding=1)
+        r_conv0 = img_conv(
+            name='stem_branch1_r_conv0',
+            input=input,
+            num_channels=160,
+            num_filters=64,
+            filter_size=1,
+            stride=1,
+            padding=0)
+        r_conv1 = img_conv(
+            name='stem_branch1_r_conv1',
+            input=r_conv0,
+            num_channels=64,
+            num_filters=64,
+            filter_size=(7, 1),
+            stride=1,
+            padding=(3, 0))
+        r_conv2 = img_conv(
+            name='stem_branch1_r_conv2',
+            input=r_conv1,
+            num_channels=64,
+            num_filters=64,
+            filter_size=(1, 7),
+            stride=1,
+            padding=(0, 3))
+        r_conv3 = img_conv(
+            name='stem_branch1_r_conv3',
+            input=r_conv2,
+            num_channels=64,
+            num_filters=96,
+            filter_size=3,
+            stride=1,
+            padding=1)
+        return paddle.layer.concat(input=[l_conv1, r_conv3])
+
+    def block2(input):
+        conv0 = img_conv(
+            name='stem_branch2_conv0',
+            input=input,
+            num_channels=192,
+            num_filters=192,
+            filter_size=3,
+            stride=2,
+            padding=1)
+        pool0 = paddle.layer.img_pool(
+            name='stem_branch2_pool0',
+            input=input,
+            num_channels=192,
+            pool_size=3,
+            stride=2,
+            pool_type=paddle.pooling.Max())
+        return paddle.layer.concat(input=[conv0, pool0])
+
+    conv3 = block0(conv2)
+    conv4 = block1(conv3)
+    conv5 = block2(conv4)
+    return conv5
+
+
+def Inception_A(input, depth):
+    b0_pool0 = paddle.layer.img_pool(
```

Comment:
I checked the TF implementation of inception-v4 (https://github.com/tensorflow/models/blob/master/research/slim/nets/inception_v4.py). In fact, this structure is the same as that in TF because the order of branches do not affect the structure. I add the branch as the Figure 4 in the original paper (https://arxiv.org/pdf/1602.07261.pdf) from left to right. However, the TF implementation does not follow the order as that in the figure. But the order of branches does not matter. The structure is in fact the same.

Reply:
I get. Sorry for this and the late reply.